src/amo/sagas/collections.js

```diff
@@ -70,10 +75,10 @@ export function* fetchCurrentCollection({
       ...baseParams,
       page,
     };
-    const { detail, addons } = yield all({
-      detail: call(api.getCollectionDetail, detailParams),
-      addons: call(api.getCollectionAddons, addonsParams),
-    });
+    const detail = yield call(api.getCollectionDetail, detailParams);
+    const addons = detail.addon_count ?
```

Comment:
Why don't you just make the `addons` parameter of `loadCurrentCollection()` optional? It probably should have always been optional; it has always been possible to load a collection without add-ons. I looked at the action and reducer and I think making it optional would be fine.

Reply:
Actually, let me back up. What is triggering the 404? Is it the API call to get add-ons? The description of your original bug made it sound like the API returns zero results but not a 404.

Reply:
I was sure that the API call to get the add-ons was triggering the 404, but now I cannot reproduce it at all, either on -dev or locally, either on master or with my branch, so I have no idea what's going on. Let's just leave this for now and if I experience it again I'll do some more digging. Sorry for the noise.